include/unifex/into_variant.hpp

```diff
@@ -0,0 +1,136 @@
+#pragma once
+
+#include <unifex/config.hpp>
+#include <unifex/async_trace.hpp>
+#include <unifex/bind_back.hpp>
+#include <unifex/receiver_concepts.hpp>
+#include <unifex/sender_concepts.hpp>
+#include <unifex/type_list.hpp>
+
+#include <tuple>
+#include <variant>
+
+#include <unifex/detail/prologue.hpp>
+
+namespace unifex {
+namespace _into_variant {
+template <typename Receiver>
+struct _receiver {
+  struct type;
+};
+template <typename Receiver>
+using receiver_t = typename _receiver<Receiver>::type;
+
+template <typename Receiver>
+struct _receiver<Receiver>::type {
+  UNIFEX_NO_UNIQUE_ADDRESS Receiver receiver_;
+
+  template(typename... Values)(
+  requires receiver_of< Receiver, Values...>)
+  void set_value(Values&&... values) && {
+    unifex::set_value((Receiver &&)(receiver_),
+        std::variant<std::tuple<remove_cvref_t<Values>...>>(
+        std::make_tuple((Values &&)(values)...)));
```

Comment:
This isn't quite right. This sends a variant with only a single alternative. The predecessor sender may invoke `set_value` with one of a set of argument types, and this is only one of those. The variant type you compute below in your sender, namely `sender_value_types_t<Predecessor, std::variant, std::tuple>`, is the type of the variant this should be forwarding to the nested receiver. Your sender with have to parameterize this receiver on that type.